command/debug/debug_state.go

```diff
@@ -0,0 +1,56 @@
+package debug
+
+import (
+	"sync"
+)
+
+var (
+	state *DebugState
+	once  sync.Once
+)
+
+// DebugState stores the variables map required for debugging vm flow.
+type DebugState struct {
+	mu  sync.Mutex
+	env map[string]map[string]string
+}
+
+func (s *DebugState) Get(stageRuntimeID string) map[string]string {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	val, ok := s.env[stageRuntimeID]
+	if ok {
+		return val
+	}
+	return nil
+}
+
+func (s *DebugState) Add(stageRuntimeID string, envs map[string]string) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if _, ok := s.env[stageRuntimeID]; !ok {
+		s.env[stageRuntimeID] = make(map[string]string)
+	}
+	for k, v := range envs {
+		s.env[stageRuntimeID][k] = v
+	}
+}
+
+func (s *DebugState) Delete(stageRuntimeID string) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	delete(s.env, stageRuntimeID)
+}
+
+func GetDebugState() *DebugState {
+	once.Do(func() {
+		state = &DebugState{
+			mu:  sync.Mutex{},
+			env: make(map[string]map[string]string),
```

Comment:
@shubham149 create a new class for storing this info. Im making the map as string->string. As later we might need to store other info as well.

Reply:
we should not create it thinking of future. Current entry would be better having just a bool. If required later, it can be transformed to a different structure.

cleanupVm: make(map[string]bool)